js/main.js

```diff
@@ -229,5 +242,66 @@ function renderNewEntry(entryObj) {
   $entryNotes.textContent = entryObj.notes;
   $entryDetailsDiv.appendChild($entryNotes);
 
+  var $entryEditButton = document.createElement('a');
+  $entryEditButton.textContent = 'EDIT/DELETE';
+  $entryEditButton.className = 'button entry-edit-button';
+  $entryEditButton.setAttribute('data-view', 'edit-entry');
+  $entryEditButton.setAttribute('href', '#');
+  $entryEditButton.setAttribute('entry-id', entryObj.entryId);
+  $entryDetailsDiv.appendChild($entryEditButton);
+
   return $entryLi;
 }
+
+var $editEntryForm = document.querySelector('.edit-entry-form');
+var $editEntryImage = document.querySelector('[data-view="edit-entry"] img');
+var $entryImageUrlEditInput = document.querySelector('#entryImageUrlEdit');
+var $titleEditInput = document.querySelector('#titleEdit');
+var $notesEditInput = document.querySelector('#notesEdit');
+var currentEntry;
```

Comment:
Given that this variable and the `currentEntryIndex` are so crucial to the functionality of your app, I think this would be great information to add to your data model in the `data` variable!

Reply:
I have added the `entryId` as a new property for every entry object in the `entries` array in the `data` variable. This `currentEntry` variable is only referring to the entry that is currently being edited/deleted, if any. Is there any reason to add that to the `data` variable?

Reply:
The reasoning would be that you want that object to be your one source of truth for the state of your application. It is already storing which view you are currently on, so a record of what the current entry and its id are might be beneficial to store in the same space. For instance, if your user last left the app on the `edit-entry` view, and you preserve that in localStorage, but you don't preserve which entry they were referencing, it would not be possible to restore their previous view. If you do store it though, the user could open the app to the same view and see the same data on a return to your application :+1:

Reply:
👍